pkgs/by-name/op/opengist/package.nix

```diff
@@ -4,38 +4,37 @@
   buildNpmPackage,
   fetchFromGitHub,
   moreutils,
-  npm-lockfile-fix,
+  jd-diff-patch,
   jq,
   git,
 }:
 let
   # finalAttrs when 🥺 (buildGoModule does not support them)
   # https://github.com/NixOS/nixpkgs/issues/273815
-  version = "1.7.5";
+  version = "1.8.2";
   src = fetchFromGitHub {
     owner = "thomiceli";
     repo = "opengist";
     rev = "v${version}";
-    hash = "sha256-mZ4j9UWdKa3nygcRO5ceyONetkks3ZGWxvzD34eOXew=";
-
-    # follow https://github.com/thomiceli/opengist/pull/350 and remove here
-    postFetch = ''
-      ${lib.getExe npm-lockfile-fix} $out/package-lock.json
-    '';
+    hash = "sha256-mx1RPDvoewSypwRRKZBImTC6119Nlz3JD212hxjNbNA=";
   };
+  jd' = lib.getExe jd-diff-patch;
+  jq' = lib.getExe jq;
+  sponge = "${moreutils}/bin/sponge";
 
   frontend = buildNpmPackage {
     pname = "opengist-frontend";
     inherit version src;
 
-    nativeBuildInputs = [
-      moreutils
-      jq
-    ];
-
     # npm complains of "invalid package". shrug. we can give it a version.
-    preBuild = ''
-      jq '.version = "${version}"' package.json | sponge package.json
+    # esbuild optional dependencies installed explicitly
+    # as they are missing for non-x86_64-linux in package-lock.json
+    # nix shell nixpkgs#{nodejs,jd-diff-patch} -c \
+    #   sh -c "npm add -D esbuild@0.18.20; git difftool -yx jd @ -- package-lock.json > package-lock-esbuild.jd.diff"
+    prePatch = ''
```

Comment:
```suggestion
    postPatch = ''
```

Reply:
I will change this as well as clean up the comments.
Maybe see if the esbuild patch could be upstreamed

Reply:
https://github.com/thomiceli/opengist/pull/393

if that gets in I will either request a 1.8.3 or package unstable version.